rfd/0159-password-status.md

```diff
@@ -0,0 +1,128 @@
+---
+authors: Bartosz Leper (bartosz.leper@goteleport.com)
+state: draft
+---
+
+# RFD 0159 - Password Status
+
+## Required Approvers
+
+- Engineering:
+- Security:
+- Product:
+
+## What
+
+We want to improve the UX of our flows related to changing passwords and management of authentication devices (both MFA and passwordless). The following new features must be supported:
+
+- The user must be able to set password having only passwordless credentials configured.
```

Comment:
Can you help explain the motivation from the users perspective?

Initially I figured the perspective of a user who started with passwordless, but their threat model has changed such that they believe they need a password to further secure their account.

However if that is the motivation of the user, it seems that potentially not requiring the password as part of a password change (or other security sensitive actions), it compromises that motivation of additional security.

Why would an account want a password set, if that password does not provide any additional protections?

Or is the flow expected such that after the initial password set the account no longer is possible to function in passwordless, thus requiring a password for future changes?

Reply:
> Why would an account want a password set, if that password does not provide any additional protections?

I think it's more about usability than additional protections. Maybe a user wants to log in from a workstation where they don't have a compatible MFA device, for example.

A user with a password can add passwordless methods at any point in time and still continue to use their password, but a user who starts with passwordless (today) can never use a password. As a result, the most flexible option is to always start with a password. We'd prefer that both are equivalent and people can start with the strongest (passwordless) option without feat of being locked-in to that option forever.

Reply:
Thank you for clarifying, that makes sense